flink-tests/src/test/java/org/apache/flink/test/streaming/api/FileReadingWatermarkITCase.java

```diff
@@ -62,8 +64,17 @@ public void testWatermarkEmissionWithChaining() throws Exception {
         JobExecutionResult result = env.execute();
 
         int actual = result.getAccumulatorResult(NUM_WATERMARKS_ACC_NAME);
-
-        assertTrue("too few watermarks emitted: " + actual, actual >= MIN_EXPECTED_WATERMARKS);
+        long expected =
```

Comment:
This computation seems a bit off for me. The `RUNTIME_ACC_NAME` is taken long after there can be watermarks generated.

`close` is called to release resources, one we closed the mailbox, all timers waited for pending checkpoints etc. All those actions might take some time, but that should not add to the expected number of watermarks.

Reply:
Good point.
I'll change it to update the accumulator for the running time when processing the last record in sink.